src/algo.rs

```diff
@@ -78,29 +77,37 @@ impl Incorporate for Vec<Profile> {
     }
 }
 
+fn min_duration(a: Option<u16>, b: Option<u16>) -> Option<u16> {
+    match (a, b) {
+        (None, _) => b,
+        (_, None) => a,
+        (Some(a), Some(b)) => Some(a.min(b)),
+    }
+}
+
 // It returns all the possible routes, from all possible nodes to the given destination
 pub fn compute(timetable: &Timetable, destinations: &[usize]) -> Vec<Vec<Profile>> {
     let mut arr_time_with_trip: Vec<_> = timetable.trips.iter().map(|_| None).collect();
     let mut profiles: Vec<_> = timetable.stops.iter().map(|_| Vec::new()).collect();
-    let mut final_footpaths = HashMap::new();
+    let mut final_footpaths: Vec<Option<u16>> = timetable.stops.iter().map(|_| None).collect();
```

Comment:
isn't it more efficient to write this
`let mut final_footpaths: = vec![timetable.stops.len(); None];`

Reply:
yes indeed! thanks